### PR TITLE
Add hotkey to new folder form #134

### PR DESCRIPTION
**Pull request by verheyenkoen**
This PR adds hotkey functionality to the folder view on the "Make new folder" button as described in #134.

### CODE REVIEW

File: views/new_folder.templ [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import (
+	"github.com/ugent-library/deliver/ctx"
+	"github.com/ugent-library/deliver/models"
+	"github.com/ugent-library/okay"
+)
+
+type NewFolderArgs struct {
+	Folder *models.Folder
+	Errors *okay.Errors
+}
```
Comment (verheyenkoen) [inline views/new_folder.templ:12]: Decided to put the args struct here and use it from `spaces.go`. Not sure if this is the right place to put it.

File: views/show_folder.templ [hunk 2]
```diff
@@ -33,9 +33,16 @@ templ ShowFolder(c *ctx.Ctx, folder *models.Folder) {
 						</div>
 						<div class="bc-toolbar-right">
 							<div class="bc-toolbar-item">
-								<a class="btn btn-link btn-link-muted" href={ templ.URL(c.Path("space", "spaceName", folder.Space.Name).String()) }>
+								<a
+									class="btn btn-link btn-link-muted"
+									href={ templ.URL(c.Path("space", "spaceName", folder.Space.Name, "focus", "new-folder").String()) }
```
Comment (verheyenkoen) [inline views/show_folder.templ:38]: The last two arguments translate to a `?focus=new-folder` query string that is picked up here:
https://github.com/ugent-library/deliver/blob/feature/new-folder-hotkey/handlers/spaces.go#L168-170

This means this is visible in the URL.

**Comment by verheyenkoen** [comment]
@mietcls One problem I identified is, if you hit the "n" hotkey while uploading files, you are redirected and all incomplete uploads are lost. Since this is also the case if you navigate away in another manner (eg. clicking a link), I decided not to fix that in the scope of this ticket. 

Instead I suggest we add a confirm step when leaving the page during uploads. This can be done by adding/removing an event handler for the [beforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) event.

**Comment by mietcls** [comment]
Thank you, works as expected!
Created a new issue based on your remark, a good enhancement: https://github.com/ugent-library/deliver/issues/142

**Review by mietcls — APPROVED** [review]
Visually and functionally OK

**Comment by verheyenkoen** [comment]
@nics Can you review and merge this?

File: views/new_folder.templ (hunk 1 above)
Comment (nics) [reply views/new_folder.templ:13]: would also have put it here

**Review by nics — APPROVED** [review]

**Pull request merged**